packages/iov-bns/src/decode.ts

```diff
@@ -328,13 +329,41 @@ function decodeProposalStatus(status: codecImpl.gov.Proposal.Status): ProposalSt
   }
 }
 
+function decodeValidators(validators: readonly codecImpl.weave.IValidatorUpdate[]): Validators {
+  const initialValidators: Validators = {};
+  return validators.reduce((result, validator) => {
+    if (!validator.pubKey || !validator.pubKey.data) {
+      throw new Error("Validator is missing pubKey data");
+    }
+    const pubkeyHex = Encoding.toHex(validator.pubKey.data);
+    return {
+      ...result,
+      [pubkeyHex]: { power: asIntegerNumber(validator.power) },
+    };
+  }, initialValidators);
+}
+
 function decodeRawProposalOption(prefix: "iov" | "tiov", rawOption: Uint8Array): ProposalAction {
   const option = codecImpl.bnsd.ProposalOptions.decode(rawOption);
   if (option.govCreateTextResolutionMsg) {
     return {
       kind: ActionKind.CreateTextResolution,
       resolution: decodeString(option.govCreateTextResolutionMsg.resolution),
     };
+  } else if (option.escrowReleaseMsg) {
+    return {
+      kind: ActionKind.ReleaseGuaranteeFunds,
+      // TODO: codecImpl.bnsd.ProposalOptions.decode currently returns a Buffer
```

Comment:
Right, this is because of the JS implementation of the protobuf encoder/decoder. The copy into Uint8Array is perfect